packages/deploy-tooling/src/base/validate.ts

```diff
@@ -227,6 +227,12 @@ async function validatePackageWithAdt(
         return;
     }
 
+    // ADT expects input package
+    let inputPackage = input.package;
+    if (inputPackage.toUpperCase() === '$TMP') {
+        inputPackage = '$TMP';
+    }
```

Comment:
I think it could make sense to upper case all package names for ABAP.
I remember correctly they are always upper case.

Reply:
Thanks @devinea. I haven't got chance to test if small case package names other than '$tmp' would work for deploy. Currently we are treating it as case sensitive check and just give exception to '$tmp'. Otherwise, we report as an invalid input, which is not informative and non-blocking.